yarn-project/p2p/src/services/reqresp/connection-sampler/connection_sampler.ts

```diff
@@ -69,32 +69,54 @@ export class ConnectionSampler {
   getPeer(excluding?: Map<string, boolean>): PeerId | undefined {
     // In libp2p getPeers performs a shallow copy, so this array can be sliced from safetly
     const peers = this.libp2p.getPeers();
+    const [peer] = this.getPeerFromList(peers, excluding);
+    return peer;
+  }
 
+  /**
```

Comment:
Updated this method to include passing in the peers such that the list can be mutated and sampled without replacement by the batch method. This means we can actually sample rather than iterating over the peer list